projects/ngqp/core/src/lib/directives/query-param.directive.ts

```diff
@@ -0,0 +1,64 @@
+import { Directive, Inject, Input, OnChanges, OnDestroy, Optional, Self, SimpleChanges } from '@angular/core';
+import { QueryParamGroupService } from './query-param-group.service';
+import { QueryParamController } from './query-param-controller.interface';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { selectValueAccessor } from '../accessors/util';
+import { QueryParam } from '../model/query-param';
+import { QueryParamGroup } from '../model/query-param-group';
+
+/**
+ * Binds a standalone {@link QueryParam} to a DOM element.
+ *
+ * This directive accepts the name of a {@link QueryParam} without having outer {@link QueryParamGroup}.
+ * It binds this parameter to the host element, which is required to have a [ControlValueAccessor]
+ * {@link https://angular.io/api/forms/ControlValueAccessor}.
+ */
+@Directive({
+    selector: '[queryParam]',
+    providers: [QueryParamGroupService],
+})
+export class QueryParamDirective implements QueryParamController, OnChanges, OnDestroy  {
+
+    /**
+     * The name of the {@link QueryParam} inside its parent {@link QueryParamGroup}.
```

Comment:
I think this entire comment is copy-pasted and wrong.

Reply:
Shame on me, fixed.